src/inference/src/cnn_network_ngraph_impl.cpp

```diff
@@ -358,45 +358,55 @@ void CNNNetworkNGraphImpl::reshape() {
-    if (inputShapes.empty())
-        return OK;
+    try {
+        if (inputShapes.empty())
+            return OK;
 
-    const auto& params = _ngraph_function->get_parameters();
+        const auto& params = _ngraph_function->get_parameters();
 
-    // Check that we need to do reshape only if input shapes will be changed
-    bool needReshape = false;
-    for (const auto& param : params) {
-        const auto it = inputShapes.find(param->get_friendly_name());
-        if (it == inputShapes.end()) {
-            continue;
-        }
-        if (param->get_output_partial_shape(0).is_dynamic() || param->get_output_partial_shape(0) != it->second) {
-            needReshape = true;
-            break;
+        // Check that we need to do reshape only if input shapes will be changed
+        bool needReshape = false;
+        for (const auto& param : params) {
+            const auto it = inputShapes.find(param->get_friendly_name());
+            if (it == inputShapes.end()) {
+                continue;
+            }
+            if (param->get_output_partial_shape(0).is_dynamic() || param->get_output_partial_shape(0) != it->second) {
+                needReshape = true;
+                break;
+            }
         }
-    }
 
-    if (!needReshape)
-        return OK;
+        if (!needReshape)
+            return OK;
 
-    // save original parameters shape
-    std::map<std::string, ngraph::PartialShape> originalInputShapes;
-    for (const auto& param : params) {
-        originalInputShapes[param->get_friendly_name()] = param->get_output_partial_shape(0);
-    }
+        // save original parameters shape
+        std::map<std::string, ngraph::PartialShape> originalInputShapes;
+        for (const auto& param : params) {
+            originalInputShapes[param->get_friendly_name()] = param->get_output_partial_shape(0);
+        }
 
-    try {
-        ngraph::pass::Manager ssr_manager;
-        ssr_manager.register_pass<ngraph::pass::SmartReshape>();
-        ssr_manager.run_passes(_ngraph_function);
+        try {
+            ngraph::pass::Manager ssr_manager;
+            ssr_manager.register_pass<ngraph::pass::SmartReshape>();
+            ssr_manager.run_passes(_ngraph_function);
 
-        reshape(inputShapes);
-    } catch (std::exception& ex) {
-        reshape(originalInputShapes);
-        return DescriptionBuffer(GENERAL_ERROR, responseDesc) << ex.what();
-    }
+            reshape(inputShapes);
+        } catch (std::exception& ex) {
+            reshape(originalInputShapes);
+            return DescriptionBuffer(GENERAL_ERROR, responseDesc) << ex.what();
+        }
 
-    return OK;
+        return OK;
+    } catch (const InferenceEngine::GeneralError&) {
+        return GENERAL_ERROR;
+    } catch (const ov::Exception&) {
+        return GENERAL_ERROR;
+    } catch (const std::out_of_range&) {
+        return OUT_OF_BOUNDS;
+    } catch (const std::runtime_error&) {
+        return GENERAL_ERROR;
+    }
```

Comment:
Do we have all these exception types? Can we also add the exception message?

Reply:
All the exceptions may be thrown from `reshape()`. Added exception message handling